docs: add style guide

jahtomini opened the pull request:
## 🛠️ Fixes issue(s)
 Closes #121 

## 👨‍💻 Change(s) proposed
- added STYLEGUIDE.md
- added recommended extensions for VSCode workspace
- updated README to include instructions for installing extensions

## ✔️ Checklist (Check all the applicable boxes) <!-- Follow the below conventions to check the box -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[ ] - Not correct; marked as **not** done
-->

- [ ] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

<!-- Add notes to reviewers if applicable -->

## 📷 Screenshots

jahtomini committed:
docs: add style guide

- add STYLEGUIDE.md
- add recommended extensions for VSCode workspace
- update README to include instructions for installing extensions

Ngoakor12 commented on frontend/STYLEGUIDE.md:
````diff
@@ -0,0 +1,127 @@
+<Button variant="secondary" />
+```
+
+<h2 id='2'>Editor Setup</h2>
````
I don't think the setup part is necessary, maybe just mention that installing the those two packages may be beneficial but also that you should run the linter to fix your linting problems

Also I think it should maybe just be added as one of the things to setup in `contributing.md` if yout it's necessary

Ngoakor12 commented on frontend/STYLEGUIDE.md:
```diff
@@ -0,0 +1,127 @@
+<Foo
+ />
+
+// good 👍🏽
```
I see you used a thumbs up :+1:  emoji for one good example and not for others. I definitely don't mind them, but I think if you're going to use them, use them everywhere and be consistent

Ngoakor12 commented on frontend/STYLEGUIDE.md:
```diff
@@ -0,0 +1,127 @@
+
+<h3 id='1.3'>Quotes</h3>
+
+Always use double quotes (") for JSX attributes, but single quotes (') for all other JS.
```
`but single quotes (') for all other JS.` seems unnecesary and confusing, plus I'm not a fan of single quotes and don't remember using them at all

Ngoakor12 commented on README.md:
````diff
@@ -81,3 +81,21 @@ npm install
 npm start
 ```
+
+#### Set up ESLint & Prettier
````
See the comment I made about the linter and prettier setup below

Ngoakor12 replied:
You forgot to update the eslint and prettier section in here(`README.md`).

- Make it match the eslint part in `CONTRIBUTING.md` or just add a link to that part of the setup in the `CONTRIBUTING.md` file
- Also I don't think the part about installing the packages any useful because all the packages will only work when you `npm install` which I'm pretty sure they would have done setting up the frontend and backend. I think the most important part of this is the recommendation to install the respective extension since they do help although not critical since running `npm run lint` takes care of the styling

Ngoakor12 reviewed (changes requested):
Mostly looks good to me :relaxed: 

...but I did leave some comments

jahtomini committed:
Moved 'editor setup' to CONTRIBUTING.md & other minor tweaks

jahtomini commented:
That all makes sense, just made some changes

Ngoakor12 commented on frontend/STYLEGUIDE.md:
```diff
@@ -0,0 +1,107 @@
+
+<h3 id='1.3'>Quotes</h3>
+
+Always use double quotes (") for JSX attributes.
```
Double quotes are used throughout. So can you please remove the "for JSX attributes." part

jahtomini replied:
Sure

Ngoakor12 commented on frontend/STYLEGUIDE.md:
````diff
@@ -0,0 +1,107 @@
+
+Always self-close tags that have no children.
+
+```javascript
+// bad
+<Button variant="primary"></Button>
+
+// good
````
you forgot to add some emojis here

jahtomini replied:
Oops, thought I had added them throughout, will fix that

Ngoakor12 reviewed (changes requested):
The styleguide part is almost there, a few things to address. I also left some more comments about other part you edited

jahtomini commented:
> You forgot to update the eslint and prettier section in here(README.md).

Don't know how that happened, will get that done once I'm back.

However, I noticed two README files in the `/frontend` directory: `README.md & README2.md`. Do these serve any purpose? One of them seems to be practically empty.

Ngoakor12 commented:
> However, I noticed two README files in the `/frontend` directory: `README.md & README2.md`. Do these serve any purpose? One of them seems to be practically empty.

You can remove the `README2.md`

Ngoakor12 commented:
> > However, I noticed two README files in the `/frontend` directory: `README.md & README2.md`. Do these serve any purpose? One of them seems to be practically empty.
> 
> You can remove the `README2.md`

Scratch that, remove the empty one

jahtomini committed:
Update ESLint & Prettier section & other minor tweaks
- run prettier with lint script
- delete empty README
- more consistency in STYLEGUIDE
- update ESLint & Prettier section in README

jahtomini committed:
Minor text edit

Ngoakor12 commented:
@jahtomini This is ready for a review right? It looks done. Or do you want to add something else? I'm asking because you didn't say anything this time

jahtomini commented:
> @jahtomini This is ready for a review right? It looks done. Or do you want to add something else? I'm asking because you didn't say anything this time

Yep, ready for review 👍🏾

Ngoakor12 reviewed (approved):
That's a wrap! another 100 file changes PR merged :sweat_smile: 

Thank you very much :ok_hand:
